Bump all modules to latest

eed3si9n opened the pull request:
This integrates latest modules and existing ripple PRs.

jvican committed:
Synchronize with changes to Zinc logger API

https://github.com/sbt/zinc/pull/325

jvican committed:
Remove any reference to `F0` and `F1`

Syncs with https://github.com/sbt/zinc/pull/359 and
https://github.com/sbt/util/pull/84.

jvican committed:
Add setting to switch to text analysis

As per suggestions in https://github.com/sbt/zinc/pull/351.

jvican committed:
Sync up with changes to `AnalysisContents`

eed3si9n committed:
An attempt to reproduce the cache update

Duhemm committed:
Adapt to changes in ConsoleAppender

eed3si9n committed:
Bump modules

eed3si9n committed:
Merge branch 'pr/3316' into wip/integration

eed3si9n committed:
Merge branch 'pr/3317' into wip/integration

eed3si9n committed:
Merge branch 'pr/3321' into wip/integration

Duhemm committed:
Adapt to new library management API

eed3si9n committed:
Adjust to upstream changes

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

typesafe-tools commented:
A dbuild validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/99/console">is in progress...</a>

typesafe-tools commented:
[//]: # (Refs: pull/3322/head-1.0-1.0-1.0-1.0-1.0.x)
dbuild <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/99/">has checked</A> the following projects against Scala 2.12:

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/pull/3322">pull/3322/head      </A> | sbt/sbt@f0ac6ae11cf472e6f6e1b294daa84df0a2fbaa23
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/1.0">1.0                 </A> | sbt/zinc@daaec42de7db636084384545015dbceeab51c7b6
  io                | <A HREF="https://github.com/sbt/io/tree/1.0">1.0                 </A> | sbt/io@49a9cea2906ecb62ea690b5d5dddc257e0ada151
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/1.0">1.0                 </A> | sbt/librarymanagement@0147e0c77578ca4852137a6113bcac4c7adeeb38
  util              | <A HREF="https://github.com/sbt/util/tree/1.0">1.0                 </A> | sbt/util@5b85002a69dfcae4429fb6034831f4766aba509b
  website           | <A HREF="https://github.com/sbt/website/tree/1.0.x">1.0.x               </A> | sbt/website@1721355e3823bc3c83dac88a2c07f7727b2f9d26

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild?sbt=pull/3322/head&zinc=1.0&io=1.0&librarymanagement=1.0&util=1.0&website=1.0.x&deploy_to_bintray=false">(restart)</A>

eed3si9n commented:
Ran scripted over night:

[error] (sbtRoot/*:scripted) Failed tests:
- [x] [error] 	actions / external-doc
- [x]  [error] 	actions / cross
- [x]  [error] 	compiler-project / separate-analysis-per-scala
- [x]  [error] 	compiler-project / error-in-invalidated
- [x]  [error] 	compiler-project / macro-config
- [x]  [error] 	compiler-project / semantic-errors
- [x]  [error] 	dependency-management / update-sbt-classifiers
- [x]  [error] 	dependency-management / cached-resolution-conflicts
- [x]  [error] 	dependency-management / pom-advanced
- [x]  [error] 	dependency-management / make-pom
- [x]  [error] 	dependency-management / cache-update
- [x]  [error] 	dependency-management / extra
- [x]  [error] 	dependency-management / cached-resolution-classifier
- [x]  [error] 	dependency-management / exclude-dependencies
- [x]  [error] 	dependency-management / pom-scope
- [x]  [error] 	project / auto-plugins
- [x]  [error] 	project / extra
- [x]  [error] 	source-dependencies / transitive-memberRef
- [x]  [error] 	source-dependencies / trait-member-modified
- [x]  [error] 	tests / scala-instance-classloader
- [x]  [error] 	tests / serial

[error] Total time: 7287 s, completed Jul 16, 2017 3:22:24 AM

eed3si9n committed:
Fixes actions/external-doc

eed3si9n committed:
Roll back Vector changes in Keys.scala

Keys should Seq type since it's fairly common to use List in build.sbt, and it would cause annoying build changes.
Fixes actions/cross

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

eed3si9n committed:
notes

eed3si9n committed:
Fixes compiler-project/separate-analysis-per-scala

eed3si9n committed:
Fixes compiler-project/error-in-invalidated

eed3si9n committed:
Fixes compiler-project/macro-config

eed3si9n committed:
Fixes compiler-project/semantic-errors

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

eed3si9n committed:
Fixes dependency-management/update-sbt-classifiers

eed3si9n committed:
Mark compiler-project/macro-config pending

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

eed3si9n committed:
Fixes dependency-management/cached-resolution-conflicts

eed3si9n committed:
Fixes dependency-management/pom-advanced

eed3si9n committed:
Fixes dependency-management/make-pom

eed3si9n committed:
Merge branch 'wip/cached-update' into wip/integration

jvican commented on sbt/src/sbt-test/compiler-project/macro-config/build.sbt:
```diff
@@ -1,24 +1,27 @@
 
 // adds standard compile, console, package tasks for src/macro/
-inConfig(config("macro"))(Defaults.configSettings)
+inConfig(Macro)(Defaults.configSettings)
```
Much better.

jvican commented on notes/1.0.0.markdown:
```diff
@@ -38,10 +37,15 @@ This is the RC-1 release of sbt 1.0.
+
+- Java classes under the `xsbti.compile` package such as `IncOptions` hides the constructor. Use the factory method `xsbti.compile.Foo.of(...)`.
+- Renames `ivyScala: IvyScala` to `scalaModuleInfo: ScalaModuleInfo`.
+- `xsbti.Reporter#log(...)` takes `xsbti.Problem` as the parameter. Call `log(problem.position, problem.message, problem.severity)` to delegate to the older `log(...)`.
```
Good catch. I'll open a ticket in Zinc to ask contributors to migrate all these tests to Zinc when possible.

eed3si9n committed:
Some helpful changes for update caching

There's a bug in sbt/util#93

eed3si9n committed:
Fixes dependency-management/cached-resolution-classifier

eed3si9n committed:
Fixes dependency-management/exclude-dependencies

eed3si9n committed:
Fixes dependency-management/pom-scope

eed3si9n committed:
Disable compiler-project/macro-config

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

typesafe-tools commented:
A dbuild validation involving this pull request <a href="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/100/console">is in progress...</a>

typesafe-tools commented:
[//]: # (Refs: pull/3322/head-1.0-1.0-1.0-1.0-1.0.x)
dbuild <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/100/">has checked</A> the following projects against Scala 2.12:

Project             | Reference            | Commit
:------------------ | :------------------- | :----------------
  sbt               | <A HREF="https://github.com/sbt/sbt/pull/3322">pull/3322/head      </A> | sbt/sbt@96b5c3bb45f4ee7147fa118fa670e8a727f019fe
  zinc              | <A HREF="https://github.com/sbt/zinc/tree/1.0">1.0                 </A> | sbt/zinc@daaec42de7db636084384545015dbceeab51c7b6
  io                | <A HREF="https://github.com/sbt/io/tree/1.0">1.0                 </A> | sbt/io@49a9cea2906ecb62ea690b5d5dddc257e0ada151
  librarymanagement | <A HREF="https://github.com/sbt/librarymanagement/tree/1.0">1.0                 </A> | sbt/librarymanagement@0147e0c77578ca4852137a6113bcac4c7adeeb38
  util              | <A HREF="https://github.com/sbt/util/tree/1.0">1.0                 </A> | sbt/util@5b85002a69dfcae4429fb6034831f4766aba509b
  website           | <A HREF="https://github.com/sbt/website/tree/1.0.x">1.0.x               </A> | sbt/website@1721355e3823bc3c83dac88a2c07f7727b2f9d26

:x: The result is: ***FAILED***
<A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild?sbt=pull/3322/head&zinc=1.0&io=1.0&librarymanagement=1.0&util=1.0&website=1.0.x&deploy_to_bintray=false">(restart)</A>

eed3si9n committed:
Fixes project/auto-plugins

eed3si9n committed:
Fixes project/extra

eed3si9n committed:
Fixes source-dependencies / transitive-memberRef

eed3si9n committed:
Fixes source-dependencies/trait-member-modified

eed3si9n committed:
Fixes tests/scala-instance-classloader

eed3si9n committed:
Minor change to tests/serial

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

eed3si9n committed:
Fixes dependency-management / extra

eed3si9n committed:
bump to Util 1.0.0-M28 and Zinc 1.0.0-X20

typesafe-tools commented:
To validate this pull request against other sbt modules, please visit <A HREF="https://jenkins.scala-sbt.org:8592/job/sbt%20validator/parambuild/?sbt=pull/3322/head&deploy_to_bintray=false">this link.</A>

dwijnand commented on sbt/src/main/scala/Import.scala:
```diff
@@ -8,6 +8,8 @@ trait Import {
   type URI = java.net.URI
   type URL = java.net.URL
 
+  implicit def Seq2Vector[T](s: Seq[T]): Vector[T] = s.toVector
```
Can someone explain to me the rational for this?
